### ISSUE TITLE
Consider proposing Fedora policy prohibiting scriptlets from writing to `/var`

### ISSUE THREAD

**Issue by bgilbert** [coreos/fedora-coreos-tracker#1067]
rpm-ostree runs package scriptlets server-side while composing an image.  Because the scriptlets are run once, rather than once per machine, they can't write machine-specific state to `/var`.  rpm-ostree prevents that during compose by mounting `/var` read-only.

From time to time an updated scriptlet will start writing to `/var` (perhaps to record the state of a config or data migration) and our composes will start failing.  So far, we've handled this by asking the package maintainer to change the offending code.  That leads to some delicate conversations, as the maintainer may not have known about the requirement and may not be able to easily test it.

If we believe that scriptlets should not generally write to `/var`, should we pursue a change to the Fedora packaging policy to make that official?  That would improve visibility for the rule within the Fedora packager community, and would allow tooling (particularly CI) to check for problems before a package update ever reaches a compose.

I'm not sure whether this would need to follow the Fedora Changes process, or whether it's solely a matter for the Fedora Packaging Committee.  However, if we do submit a Change, policy [recommends](https://docs.fedoraproject.org/en-US/program_management/changes_policy/#_fedora_packaging_committee) that packaging changes go to FPC _before_ a Change proposal is submitted.

**Comment by cgwalters** [comment]
Yep, SGTM.

**Comment by dustymabe** [comment]
> If we believe that scriptlets should not generally write to `/var`, should we pursue a change to the Fedora packaging policy to make that official?

Absolutely! There are other things too, like lua support that we should either implement compatibility for or try to take them to Fedora and get a policy put in place.

**Comment by travier** [comment]
We discussed this in the [community meeting today](https://lists.fedoraproject.org/archives/list/coreos@lists.fedoraproject.org/thread/KWZOD327I54STWS2FCCNL6CN6IT7UA62/):

```
17:14:50 <travier> #agreed We will pursue a Fedora packaging policy change to
                   strongly recommend that scriplet do not write content in /var
```

**Comment by jlebon** [comment]
I opened https://pagure.io/packaging-committee/pull-request/1333.

**Comment by travier** [comment]
I've written a draft in https://hackmd.io/YdHwHwJeS-WKb_6McBBAtw?view to clarify what's `rpm-ostree` "specific" and what's generally applicable to "offline" (i.e. containers and image builds).

**Comment by jlebon** [comment]
> I've written a draft in [hackmd.io/YdHwHwJeS-WKb_6McBBAtw?view](https://hackmd.io/YdHwHwJeS-WKb_6McBBAtw?view) to clarify what's `rpm-ostree` "specific" and what's generally applicable to "offline" (i.e. containers and image builds).

Is the idea with this that we would update that packaging-committee MR with that?

**Comment by travier** [comment]
I've been asked to clarify what's rpm-ostree specific / missing features from what's generally not working already (what we 're documenting).

Maybe we create a page in the docs somewhere to put that info (and link to it here) to help with the discussion.
